In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
pd.set_option('display.max_columns', None)

In [17]:
file_path = "../data/homicidios.xlsx"
df = pd.read_excel(file_path, sheet_name="HECHOS")


Realizaremos el proceso de limpieza teniendo en cuenta las situaciones más comunes:

1. Datos faltantes en algunas celdas
2. Columnas irrelevantes (que no responden al problema que queremos resolver)
3. Registros (filas) repetidos
4. Valores extremos (*outliers*) en el caso de las variables numéricas.

Al final de este proceso de limpieza deberíamos tener un set de datos **íntegro**, listo para la fase de Análisis Exploratorio.

In [5]:
print(df.shape)
df.head()

(696, 21)


ID  N_VICTIMAS      FECHA  AAAA  MM  DD      HORA HH  \
0  2016-0001           1 2016-01-01  2016   1   1  04:00:00  4   
1  2016-0002           1 2016-01-02  2016   1   2  01:15:00  1   
2  2016-0003           1 2016-01-03  2016   1   3  07:00:00  7   
3  2016-0004           1 2016-01-10  2016   1  10  00:00:00  0   
4  2016-0005           1 2016-01-21  2016   1  21  05:20:00  5   

                             LUGAR_DEL_HECHO TIPO_DE_CALLE             Calle  \
0  AV PIEDRA BUENA Y AV FERNANDEZ DE LA CRUZ       AVENIDA  PIEDRA BUENA AV.   
1          AV GRAL PAZ  Y AV DE LOS CORRALES      GRAL PAZ    PAZ, GRAL. AV.   
2                         AV ENTRE RIOS 2034       AVENIDA    ENTRE RIOS AV.   
3      AV LARRAZABAL Y GRAL VILLEGAS CONRADO       AVENIDA    LARRAZABAL AV.   
4  AV SAN JUAN Y PRESIDENTE LUIS SAENZ PEÑA        AVENIDA      SAN JUAN AV.   

   Altura                                Cruce  \
0     NaN  FERNANDEZ DE LA CRUZ, F., GRAL. AV.   
1     NaN                  DE LOS CORRALES AV.   
2  2034.0                                  NaN   
3     NaN             VILLEGAS, CONRADO, GRAL.   
4     NaN              SAENZ PE?A, LUIS, PRES.   

                               Dirección Normalizada  COMUNA  \
0  PIEDRA BUENA AV. y FERNANDEZ DE LA CRUZ, F., G...       8   
1               PAZ, GRAL. AV. y DE LOS CORRALES AV.       9   
2                                ENTRE RIOS AV. 2034       1   
3          LARRAZABAL AV. y VILLEGAS, CONRADO, GRAL.       8   
4            SAN JUAN AV. y SAENZ PEÑA, LUIS, PRES.       1   

                                 XY (CABA)         pos x         pos y  \
0    Point (98896.78238426 93532.43437792)  -58.47533969  -34.68757022   
1    Point (95832.05571093 95505.41641999)  -58.50877521  -34.66977709   
2   Point (106684.29090040 99706.57687843)  -58.39040293  -34.63189362   
3    Point (99840.65224780 94269.16534422)  -58.46503904  -34.68092974   
4  Point (106980.32827929 100752.16915795)  -58.38718297  -34.62246630   

    PARTICIPANTES VICTIMA    ACUSADO  
0       MOTO-AUTO    MOTO       AUTO  
1  AUTO-PASAJEROS    AUTO  PASAJEROS  
2       MOTO-AUTO    MOTO       AUTO  
3         MOTO-SD    MOTO         SD  
4  MOTO-PASAJEROS    MOTO  PASAJEROS

In [6]:
duplicados_totales = df.duplicated().sum()
duplicados_totales

0

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 696 entries, 0 to 695
Data columns (total 21 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   ID                     696 non-null    object        
 1   N_VICTIMAS             696 non-null    int64         
 2   FECHA                  696 non-null    datetime64[ns]
 3   AAAA                   696 non-null    int64         
 4   MM                     696 non-null    int64         
 5   DD                     696 non-null    int64         
 6   HORA                   696 non-null    object        
 7   HH                     696 non-null    object        
 8   LUGAR_DEL_HECHO        696 non-null    object        
 9   TIPO_DE_CALLE          696 non-null    object        
 10  Calle                  695 non-null    object        
 11  Altura                 129 non-null    float64       
 12  Cruce                  525 non-null    object        
 13  Direc

Los datos no están completos, pues no todas las columnas tienen la misma cantidad de registros.

El número total de registros debería ser 696. Sin embargo las columnas "Altura", "Cruce", "Dirección Normalizada" y "Calle" tienen datos faltantes.

Por ser un set de datos reducido, optaremos por imputar las filas correspondientes.

Revisaremos en primer lugar los registros faltantes en "Calle" y "Dirección Normalizada" para ver que ocurre en esos casos, considerando que es una cantidad pequeña.

In [18]:
# Filtrar registros donde la columna "Calle" es nula
calle_nula = df[df['Calle'].isnull()]

# Filtrar registros donde la columna "Direccion normalizada" es nula
direccion_nula = df[df['Dirección Normalizada'].isnull()]

In [19]:
calle_nula

ID  N_VICTIMAS      FECHA  AAAA  MM  DD      HORA  HH  \
119  2016-0151           1 2016-11-18  2016  11  18  20:35:00  20   

    LUGAR_DEL_HECHO TIPO_DE_CALLE Calle  Altura Cruce Dirección Normalizada  \
119              SD         CALLE   NaN     NaN   NaN                   NaN   

     COMUNA    XY (CABA) pos x pos y PARTICIPANTES VICTIMA ACUSADO  
119       0  Point (. .)     .     .     PEATON-SD  PEATON      SD

Al ser solo un registro que cuenta con varios campos sin especificar, se decide eliminarlo del dataset

In [20]:
df = df[df['ID'] != '2016-0151']

In [23]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 695 entries, 0 to 695
Data columns (total 21 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   ID                     695 non-null    object        
 1   N_VICTIMAS             695 non-null    int64         
 2   FECHA                  695 non-null    datetime64[ns]
 3   AAAA                   695 non-null    int64         
 4   MM                     695 non-null    int64         
 5   DD                     695 non-null    int64         
 6   HORA                   695 non-null    object        
 7   HH                     695 non-null    object        
 8   LUGAR_DEL_HECHO        695 non-null    object        
 9   TIPO_DE_CALLE          695 non-null    object        
 10  Calle                  695 non-null    object        
 11  Altura                 129 non-null    float64       
 12  Cruce                  525 non-null    object        
 13  Dirección 

In [21]:
direccion_nula

ID  N_VICTIMAS      FECHA  AAAA  MM  DD      HORA  HH  \
38   2016-0052           1 2016-04-20  2016   4  20  20:00:00  20   
106  2016-0136           1 2016-10-25  2016  10  25  00:00:00   0   
119  2016-0151           1 2016-11-18  2016  11  18  20:35:00  20   
180  2017-0050           2 2017-04-28  2017   4  28  11:08:08  11   
181  2017-0051           1 2017-05-01  2017   5   1  03:47:47   3   
313  2018-0039           1 2018-04-21  2018   4  21  22:15:00  22   
546  2020-0026           1 2020-05-17  2020   5  17  06:40:00   6   
621  2021-0023           1 2021-03-01  2021   3   1  09:20:00   9   

                             LUGAR_DEL_HECHO TIPO_DE_CALLE  \
38                AUTOPISTA LUGONES PK 10000     AUTOPISTA   
106         AU BUENOS AIRES - LA PLATA KM. 4     AUTOPISTA   
119                                       SD         CALLE   
180  AU PERITO MORENO Y RAMAL ENLACE AU1/AU6     AUTOPISTA   
181                       AU DELLEPIANE 2400     AUTOPISTA   
313                 AUTOPISTA LUGONES KM 4.7     AUTOPISTA   
546             LUGONES, LEOPOLDO AV. KM 6,1     AUTOPISTA   
621         AU BUENOS AIRES LA PLATA  KM 4,5     AUTOPISTA   

                                    Calle  Altura Cruce Dirección Normalizada  \
38                  LUGONES, LEOPOLDO AV.     NaN   NaN                   NaN   
106     AUTOPISTA BUENOS AIRES - LA PLATA     NaN   NaN                   NaN   
119                                   NaN     NaN   NaN                   NaN   
180               AUTOPISTA PERITO MORENO     NaN   NaN                   NaN   
181  AUTOPISTA DELLEPIANE LUIS TTE. GRAL.     NaN   NaN                   NaN   
313                 LUGONES, LEOPOLDO AV.     NaN   NaN                   NaN   
546                 LUGONES, LEOPOLDO AV.     NaN   NaN                   NaN   
621     AUTOPISTA BUENOS AIRES - LA PLATA     NaN   NaN                   NaN   

     COMUNA    XY (CABA) pos x pos y     PARTICIPANTES VICTIMA      ACUSADO  
38       13  Point (. .)     .     .           MOTO-SD    MOTO           SD  
106       4  Point (. .)     .     .       MOTO-CARGAS    MOTO       CARGAS  
119       0  Point (. .)     .     .         PEATON-SD  PEATON           SD  
180       9  Point (. .)     .     .       MOTO-CARGAS    MOTO       CARGAS  
181       7  Point (. .)     .     .         AUTO-AUTO    AUTO         AUTO  
313      14  Point (. .)     .     .       PEATON-AUTO  PEATON         AUTO  
546      14  Point (. .)     .     .  MOTO-OBJETO FIJO    MOTO  OBJETO FIJO  
621       4  Point (. .)     .     .       MOTO-CARGAS    MOTO       CARGAS

Se observó que la columna "LUGAR_DEL_HECHO" contiene una combinacion de la información de las columnas "Altura", "Cruce", "Calle" y en ocasiones alguna información extra no muy relevante. Se decide eliminar esta columna ya que para los fines del proyecto se desean los datos separados para poder evaluar relaciones y hacer interpretaciones.

In [11]:
df = df.drop('LUGAR_DEL_HECHO', axis=1)